services/visual_recognition/v3.js

```diff
@@ -62,301 +37,329 @@ module.exports = function (RED) {
+          text:'Delete All : call to listClassifiers failed'});
+        node.error(err, msg);   
+      } else {
+        // Array to hold async tasks
+        var asyncTasks = [], nbTodelete = 0, nbdeleted = 0;
+        nbTodelete = body.classifiers.length;
+        body.classifiers.forEach(function (aClassifier) {
+          asyncTasks.push(function (cb) {
+            var parms = {};
+
+            parms.classifier_id = aClassifier.classifier_id;
+            node.service.deleteClassifier(parms, function(err) {
+              if (err) {
+                node.error(err, msg);
+                return cb('error');
+              }
+              nbdeleted++;
+              cb(null,parms.classifier_id);
+            });  
+          });
+        });      
+        async.parallel(asyncTasks, function(error, deletedList){
+          if (deletedList.length === nbTodelete) {
+            msg.payload = 'see msg.result.error';
+            msg.result = 'All custom classifiers have been deleted.';
+          } else {
+            msg.payload = 'see msg.result.error';
+            msg.result = 'Some Classifiers could have not been deleted;' +
+            'See log for errors.';
+          }
+          node.send(msg);
+          node.status({});           
+        });
+      }
+    }); 
+  }
 
-        if (err != null && body==null)
-        {
-          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
-          msg.result = {};
-          msg.result['error_code']= err.code;
-          if (!err.error)
-            msg.result['error']= err.error;
-          console.log('Error:', err.code);
-          node.error('Error code : ' +  err.code);
-          return;
-        }
-        else {
-          msg.result = {};
-          msg.result['all'] = body;
-          msg.payload='see msg.result'; // to remove any Buffer that could remains
-          node.send(msg); 
-          node.status({});
-        }
-      }; // actionComplete2
+  function executeService(feature, params, node, msg) {
+    switch(feature) {
+    case 'classifyImage':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.classify(params, function(err, body) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'detectFaces':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.detectFaces(params, function(err, body) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    case 'recognizeText':
+      prepareParamsCommon(params, node, msg, function () {
+        node.service.recognizeText(params, function(err, body) {
+          processResponse(err,body,feature,node,msg);
+        });
+      });
+      break;
+    }
+  }
 
-      var actionCompleteDeleteClassifier = function(err, body, other) {
+  function executeUtilService(feature, params, node, msg) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'executeUtilService' has a complexity of 6.](https://www.codacy.com/app/node-red-nodes-watson/node-red-node-watson/file/2737057949/issues/source?bid=3225115&fileBranchId=3352493#l287)